components/TextEditor/TextEditorContainer.tsx

```diff
@@ -1,30 +1,60 @@
-import React, {FunctionComponent, MutableRefObject, useRef, useState} from "react";
+import React, {FormEvent, FunctionComponent, MutableRefObject, useRef, useState} from "react";
 import Editor from "@monaco-editor/react";
 import {TextEditorSettings} from "./TextEditorSettings";
 import scss from "../TextEditor/TextEditorContainer.module.scss";
 import Link from "next/link";
+import {addTestSubmisson} from "../../pages/api/candidateApiClient.module";
+import {useRouter} from "next/router";
+import {Response} from "node-fetch";
+
 type EditorContentGetter = () => string;
+type FormStatus = "READY" | "SUBMITTING" | "ERROR" | "FINISHED"
+
 
 interface TextEditorContainerProps {
     height: string;
     width: string;
     defaultText: string;
+    token:string;
+    testNumber:number
 }
-const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText}) => {
+const TextEditorContainer: FunctionComponent<TextEditorContainerProps> = ({height, width, defaultText,token,testNumber}) => {
     const [isEditorReady, setIsEditorReady] = useState(false);
+    const [status, setStatus] = useState<FormStatus>("READY");
+    const [tokenId, setToken] = useState("");
+    const [testId, setTest] = useState(0);
+    const [testAnswer, setTestAnswer] = useState("");
+    const [error, setError] = useState("");
     const getEditorContentIfMountedRef: MutableRefObject<EditorContentGetter> = useRef(() => "");
-
+    const router=useRouter();
+    
     function handleIsEditorMounted(_getEditorContents: EditorContentGetter): void {
         setIsEditorReady(true);
         getEditorContentIfMountedRef.current = _getEditorContents;
+        setTestAnswer(getEditorContentIfMountedRef.current());
```

Comment:
won't this just set the `testAnswer` to whatever it is when you open the page?
How does it update to get the new value when you are ready to submit?

Reply:
getting the getEditorContentIfMountedRef.current() in the submitForm method now